File: src/levels/L19-event-bus/ts/base_analytics.ts
// Cada servicio que produce algo llama a mano a cada servicio interesado.
class Orders {
  // region: Orders#place
  place(order: Event) {
    new Inventory().onEvent(order)
    new Billing().onEvent(order)
    new Loyalty().onEvent(order)
    new Analytics().onEvent(order) // ← cambio 1
  }
  // endregion
}

class Returns {
  // Devoluciones copió la lista… y se le olvidó Facturación y Lealtad.
  // region: Returns#process
  process(returnRequest: Event) {
    new Inventory().onEvent(returnRequest)
    new Analytics().onEvent(returnRequest) // ← cambio 2
  }
  // endregion
}

File: src/levels/L19-event-bus/base_analytics.rb
# Cada servicio que produce algo llama a mano a cada servicio interesado.
class Orders
  # region: Orders#place
  def place(order)
    Inventory.new.on_event(order)
    Billing.new.on_event(order)
    Loyalty.new.on_event(order)
    Analytics.new.on_event(order) # ← cambio 1
  end
  # endregion
end

class Returns
  # Devoluciones copió la lista… y se le olvidó Facturación y Lealtad.
  # region: Returns#process
  def process(return_request)
    Inventory.new.on_event(return_request)
    Analytics.new.on_event(return_request) # ← cambio 2
  end
  # endregion
end
